internal/pkg/service/appsproxy/proxy/apphandler/authproxy/selector/selector.go

```diff
@@ -0,0 +1,273 @@
+package selector
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/util"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/config"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/dataapps/api"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/dataapps/auth/provider"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/proxy/pagewriter"
+	svcErrors "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const (
+	providerCookie             = "_oauth2_provider"
+	providerQueryParam         = "provider"
+	continueAuthQueryParam     = "continue_auth"
+	callbackQueryParam         = "rd" // value match OAuth2Proxy internals and shouldn't be modified (see AppDirector there)
+	selectionPagePath          = config.InternalPrefix + "/selection"
+	SignOutPath                = config.InternalPrefix + "/sign_out"
+	proxyCallbackPath          = config.InternalPrefix + "/callback"
+	ignoreProviderCookieCtxKey = ctxKey("ignoreProviderCookieCtxKey")
+	selectorHandlerCtxKey      = ctxKey("selectorHandlerCtxKey")
+)
+
+type ctxKey string
+
+type Selector struct {
```

Comment:
Moved under `authproxy` manager. Selector is able to selected between multiple oidc providers and also accorss multiple `authproxy` providers. Currently supports basic auth provisioning.

Reply:
I went into that commit and dont see `selector.go` under `oidcproxy` package. Probably should refactor directory structure, then move file, then adjust the file. :thinking:

Reply:
Fixed